### PR TITLE
Add Frogatto & Friends

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/frogatto/frogatto/issues/584
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/frogatto/frogatto/pull/580 https://github.com/frogatto/frogatto/pull/581 https://github.com/frogatto/frogatto/pull/582 https://github.com/frogatto/frogatto/pull/583 https://github.com/frogatto/frogatto/pull/585

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.frogatto.Frogatto

File: frogatto.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>frogatto.desktop</id>
```
Comment (hfiguiere) [inline frogatto.appdata.xml:3]: you should use the app-id here.

File: frogatto.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Frogatto
+Comment=2D platform game starring a quixotic frog
+Exec=frogatto
+Icon=frogatto
```
Comment (hfiguiere) [inline frogatto.desktop:5]: and use the proper icon name.

File: com.frogatto.Frogatto.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      - ./b2 install -j $FLATPAK_BUILDER_N_JOBS
+    sources:
+      - type: archive
+        url: https://boostorg.jfrog.io/artifactory/main/release/1.67.0/source/boost_1_67_0.tar.bz2
+        sha256: 2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba
+
+  - name: frogatto
+    buildsystem: simple
+    build-options:
+      - cflags: -std=c++11
```
Comment (hfiguiere) [inline com.frogatto.Frogatto.yaml:39]: are you sure? `-std=c++11` belong to `cxxflags`

File: com.frogatto.Frogatto.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+app-id: com.frogatto.Frogatto
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: frogatto
+rename-desktop-file: frogatto.desktop
+rename-appdata-file: frogatto.appdata.xml
+rename-icon: frogatto
```
Comment (hfiguiere) [inline com.frogatto.Frogatto.yaml:8]: as stated in other PR. if you do provide these files, you should install them with the right name from the ground up. The "rename" stuff is really only for when upstream doesn't conform to XDG standards.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
And please fix the upstream PR to use a proper app-id a filenames.

**Comment by Mailaender** [comment]
bot, build com.frogatto.Frogatto

File: frogatto.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.frogatto.Frogatto.desktop</id>
+  <metadata_license>CC-0</metadata_license>
+  <project_license>Zlib and proprietary</project_license>
```
Comment (hfiguiere) [inline frogatto.appdata.xml:5]: these licenses identifiers are invalid.

File: frogatto.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.frogatto.Frogatto</id>
+  <metadata_license>CC-0</metadata_license>
+  <project_license>Zlib</project_license>
+  <project_license>CC-BY-3.0</project_license>
+  <project_license>LicenseRef-proprietary</project_license>
```
Comment (hfiguiere) [inline frogatto.appdata.xml:7]:
```suggestion
  <project_license>Zlib AND CC-BY-3.0 AND LicenseRef-proprietary</project_license>
```

**Comment by Mailaender** [comment]
bot, build com.frogatto.Frogatto

**Comment by Mailaender** [comment]
bot, build com.frogatto.Frogatto

File: com.frogatto.Frogatto.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --persist=.frogatto
+
+modules:
+  - shared-modules/glew/glew.json
+  - shared-modules/glu/glu-9.json
+
+  - shared-modules/SDL/SDL-1.2.15.json
```
Comment (barthalion) [inline com.frogatto.Frogatto.yaml:18]: Have you tested sdl12-compat?

File: com.frogatto.Frogatto.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      - ./b2 install -j $FLATPAK_BUILDER_N_JOBS
+    sources:
+      - type: archive
+        url: https://boostorg.jfrog.io/artifactory/main/release/1.67.0/source/boost_1_67_0.tar.bz2
+        sha256: 2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba
+
+  - name: frogatto
+    buildsystem: simple
+    build-options:
+      - cxxflags: -std=c++11
```
Comment (barthalion) [inline com.frogatto.Frogatto.yaml:36]:
```suggestion
    build-options:
      cxxflags: -std=c++11
```

File: com.frogatto.Frogatto.yaml (hunk 7 above)
Comment (barthalion) [reply com.frogatto.Frogatto.yaml:18]: I just did, and it doesn't work.

**Pull request closed without merging**